core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -6594,6 +6594,64 @@ class ReplicaManagerTest {
       ))
     }
   }
+
+  @Test
+  def testReplicaManagerDontGenerateEmptyTopicPartition(): Unit = {
```

Comment:
It might be easier to test this in `DescribeLogDirsRequestTest` by setting up 2 topics with 2 topic partitions each and querying for both partitions of either one of the topics.

Reply:
Because there is a test that direct to test `describeLogDirs` method in ReplicaManagerTest, So I consider that to wirte a new test in `ReplicaManagerTest` is better than `DescribeLogDirsRequestTest`